src/main/scala/Ch4.scala

```diff
@@ -70,29 +70,35 @@ object Ch4Ex1 {
   // import io.chymyst.ch._
   // def p4Fmap[A, B](f: A => B): P4Data[A] => P4Data[B] = implement
   // not pass: def p4ContraFmap[A, B](f: B => A): P4Data[A] => P4Data[B] = implement
-//-type P4Data[A] = Option[A => String] => Option[A => Int] => Int
-//-
-//-def p4Fmap[A, B](f: A => B): P4Data[A] => P4Data[B] = { optAToString => optAToInt => i =>
-//-  // optAToString: Option[A => String]
-//-  // optAToInt: Option[A => Int]
-//-  // i: int
-//-  val dB: Option[B ⇒ String] ⇒ Option[B ⇒ Int] ⇒ Int = { optBToString => optBToInt =>
-//-    // optBToString: Option[B => String]
-//-    // optBToInt: Option[B => Int]
-//-
-//-
-//-    // Must return integer
-//-  }
-//-  dB
-//-}
+  type P4Data[A] = Option[A => String] => Option[A => Int] => Int
```

Comment:
Maybe it will help if you denote the common structure `Option[A => B]` as another parameterized type and define the appropriate transformation for it.

Reply:
I sorta had tried that, but I couldn't figure out how to combine them to implement a functor. For example:
```
type P4Data[A] = OptAToC[A, String] => OptAToC[A, Int] => Int
type OptAToC[A, C] = Option[A => C]
def contraFmapOptAToC[A, B, C](f: B => A): OptAToC[A, C] => OptAToC[B, C] = {
  case Some(aToC) => Some(b => aToC(f(b)))
  case _ => None
}
```
OptAToC[A, String] is a contraFunctor, but I think OptAToC[A, Int] => Int is a functor, and I'm not sure how to combine those two to make a functor, as slide 17 indicates there is a way